pkg/server/commands/batch_check_command_test.go

```diff
@@ -0,0 +1,230 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/oklog/ulid/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/internal/cachecontroller"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/internal/graph"
+	mockstorage "github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func TestBatchCheckCommand(t *testing.T) {
+	maxChecks := uint32(50)
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+	ds := mockstorage.NewMockOpenFGADatastore(mockController)
+
+	mockCheckResolver := graph.NewMockCheckResolver(mockController)
+	model := testutils.MustTransformDSLToProtoWithID(`
+		model
+			schema 1.1
+		type user
+		type doc
+			relations
+				define viewer: [user]
+	`)
+	ts, err := typesystem.NewAndValidate(context.Background(), model)
+	require.NoError(t, err)
+
+	cmd := NewBatchCheckCommand(
+		ds,
+		mockCheckResolver,
+		ts,
+		WithBatchCheckMaxChecksPerBatch(maxChecks),
+		WithBatchCheckCommandCacheController(cachecontroller.NewNoopCacheController()),
+	)
+
+	t.Run("calls_check_once_for_each_tuple_in_batch", func(t *testing.T) {
+		numChecks := 50
+		checks := make([]*openfgav1.BatchCheckItem, numChecks)
+		for i := 0; i < numChecks; i++ {
+			checks[i] = &openfgav1.BatchCheckItem{
+				TupleKey: &openfgav1.CheckRequestTupleKey{
+					Object:   "doc:doc1",
+					Relation: "viewer",
+					User:     "user:justin",
+				},
+				CorrelationId: fmt.Sprintf("fakeid%d", i),
+			}
+		}
+
+		mockCheckResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).
+			Times(numChecks).
+			Return(nil, nil)
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		result, _, err := cmd.Execute(context.Background(), params)
```

Comment:
**Question:** Are there any assertions that can be made against metadata?

Reply:
The only piece of metadata is the query count which is `0` in the unit test runner, but I can at least assert that the key and value are in there to catch and possible future modifications